File: fortran/mpiprog.f90
! File: mpiprog.f90
! Name: D.Saravanan
! Date: 08/11/2022
! MPI program working example

program parallel
include "mpif.h"
integer ierr, rank, size

! Initializes after program start,
! including program arguments
call MPI_INIT(ierr)

! Gets the rank number of the process
call MPI_COMM_RANK(MPI_COMM_WORLD, rank, ierr)

! Gets the number of ranks in the program
! determined by the mpiexec command
call MPI_COMM_SIZE(MPI_COMM_WORLD, size, ierr)

print *, "Hello world from process ", rank, " of ", size

! Finalizes MPI to synchronize ranks and then exists
call MPI_FINALIZE(ierr)

end program

File: julia/mpiprog.jl
#!/usr/bin/env julia
# File: mpiprog.jl
# Name: D.Saravanan
# Date: 08/11/2022

""" MPI program working example """

import MPI

function main()
    
    # Initializes after program start
    MPI.Init()

    # MPI.COMM_WORLD is the communicator
    comm = MPI.COMM_WORLD

    # Gets the rank number of the process
    rank = MPI.Comm_rank(comm)

    # Gets the number of ranks in the program
    # determined by the mpiexec command
    size = MPI.Comm_size(comm)

    println("rank $rank of nproc $size\n")
    
    # Finalizes MPI to synchronize ranks and then exits
    MPI.Finalize()

end

main()
